packages/@webex/plugin-cc/src/services/HttpRequest.ts

```diff
@@ -0,0 +1,156 @@
+   * It sends the request and then listens for the Welcome event
+   * If the Welcome event is received, it resolves the promise
+   * If the Welcome event is not received, it rejects the promise
+   */
+  public async subscribeNotifications(options: {body: SubscribeRequest}): Promise<WelcomeEvent> {
+    try {
+      const {body} = options;
+      const eventType = CC_EVENTS.WELCOME;
+      const subscribeResponse: SubscribeResponse = await this.webex.request({
+        service: WCC_API_GATEWAY,
+        resource: SUBSCRIBE_API,
+        method: HTTP_METHODS.POST,
+        body,
+      });
+
+      return new Promise((resolve, reject) => {
+        const timeoutId = setTimeout(() => {
+          console.error('Timeout waiting for event');
+          this.eventHandlers.delete(eventType);
+          reject(new Error('Timeout waiting for event'));
+        }, WEBSOCKET_EVENT_TIMEOUT);
+
+        // Store the event handler
+        this.eventHandlers.set(eventType, (data: any) => {
+          clearTimeout(timeoutId);
+          this.eventHandlers.delete(eventType);
+          resolve(data);
+        });
+
+        this.webSocket.connectWebSocket({
+          webSocketUrl: subscribeResponse.body.webSocketUrl,
+          subscriptionId: subscribeResponse.body.subscriptionId,
+        });
+      });
+    } catch (error) {
+      this.webex.logger.error(`Error sending service request: ${error}`);
+      throw new Error(`Error while performing service request: ${error.message}`);
+    }
+  }
+
+  /* This sends a request and waits for a websocket event
+   * It sends the request and then listens for the event type specified in the options
+   * If the event type is received, it resolves the promise
+   * If the event type is not received, it rejects the promise
+   */
+  public async sendRequestWithEvent(options: {
+    service: string;
+    resource: string;
+    method: HTTP_METHODS;
+    payload: object;
+    eventType: string;
+    success?: string[];
+    failure?: string[];
+  }): Promise<any> {
+    try {
+      const {service, resource, method, payload, eventType, success, failure} = options;
+
+      // Send the service request
+      const response = await this.webex.request({
+        service,
+        resource,
+        method,
+        body: payload,
+      });
+      this.webex.logger.log(`Service request sent successfully: ${response}`);
+
+      // Listen for the event
+      return new Promise((resolve, reject) => {
+        const timeoutId = setTimeout(() => {
+          console.error('Timeout waiting for event');
+          this.eventHandlers.delete(eventType);
+          reject(new Error('Timeout waiting for event'));
+        }, WEBSOCKET_EVENT_TIMEOUT);
+
+        // Store the event handler
+        this.eventHandlers.set(eventType, (data: any) => {
+          clearTimeout(timeoutId);
+          this.eventHandlers.delete(eventType);
+          if (success?.includes(data.type)) {
```

Comment:
Why is success and failure optional? Aren't we going to set the events for all the APIs